File: src-tauri/src/application/dto/export.rs
use std::path::PathBuf;

use serde::{Deserialize, Serialize};

use crate::domain::common::ids::{LanguageCode, PackId, PreviewToken, WorkspaceId};
use crate::domain::common::issue::ValidationIssue;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PreviewExportBundleInput {
    pub workspace_id: WorkspaceId,
    pub pack_ids: Vec<PackId>,
    pub export_language: LanguageCode,
    pub output_dir: PathBuf,
    pub output_name: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct ExportPreviewDto {
    pub pack_count: usize,
    pub card_count: usize,
    pub main_image_count: usize,
    pub field_image_count: usize,
    pub script_count: usize,
    pub warning_count: usize,
    pub error_count: usize,
    pub issues: Vec<ValidationIssue>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ExecuteExportBundleInput {
    pub preview_token: PreviewToken,
}

File: src/shared/contracts/export.ts
import type { LanguageCode, PackId, PreviewResult, PreviewToken, ValidationIssue, WorkspaceId } from "./common";
import type { JobAccepted } from "./job";

export interface PreviewExportBundleInput {
  workspaceId: WorkspaceId;
  packIds: PackId[];
  exportLanguage: LanguageCode;
  outputDir: string;
  outputName: string;
}

export interface ExportPreview {
  pack_count: number;
  card_count: number;
  main_image_count: number;
  field_image_count: number;
  script_count: number;
  warning_count: number;
  error_count: number;
  issues: ValidationIssue[];
}

export interface ExecuteExportBundleInput {
  previewToken: PreviewToken;
}

export type ExportPreviewResult = PreviewResult<ExportPreview>;
export type ExecuteExportBundleResult = JobAccepted;
